apps/balancer-tools/src/app/poolsimulator/(components)/SwapCurve.tsx

```diff
@@ -126,6 +142,47 @@ export function StableCurve() {
     ),
   ];
 
+  function getGraphScale({
+    initialAmountsIn,
+    customAmountsIn,
+    initialAmountsOut,
+    customAmountsOut,
+  }: {
+    initialAmountsIn: number[];
+    customAmountsIn: number[];
+    initialAmountsOut: number[];
+    customAmountsOut: number[];
+  }) {
+    const maxOfIn = {
+      initial: Math.max(...initialAmountsIn),
+      custom: Math.max(...customAmountsIn),
+    };
+    const minOfOut = {
+      initial: Math.min(...initialAmountsOut),
+      custom: Math.min(...customAmountsOut),
+    };
+
+    const limits = {
+      lowerIn: Math.min(maxOfIn.initial, maxOfIn.custom),
+      higherOut: Math.max(minOfOut.initial, minOfOut.custom),
+    };
+
+    if (maxOfIn.initial === maxOfIn.custom) {
```

Comment:
Did you use this condition for the case of two metastable pools, right? If so, this will not work for all the cases.

Reply:
e.g. two metastable pools but with different tokens balances.